fastactivity.php

```diff
@@ -246,11 +246,14 @@ function fastactivity_civicrm_links($op, $objectName, $objectId, &$links, &$mask
   $replace_search_links = (bool) CRM_Fastactivity_Settings::getValue('fastactivity_replace_search');
   if ($replace_search_links && $op == 'activity.selector.row') {
     foreach ($links as &$link) {
-      if ($link['name'] == 'View') {
-        $link['url'] = 'civicrm/fastactivity/view';
-      }
-      elseif ($link['name'] == 'Edit') {
-        $link['url'] = 'civicrm/fastactivity/add';
+      // GP-31270 Wrong contribution activity link (avoid replacement with fastactivity link when object leads to contribution pages)
+      if (strpos($link['url'], '/contribution') === false) {
```

Comment:
There are a few other scenarios where the link is changed from a generic activity link to something else (e.g. "Event Registration" activities). It would make more sense to reverse the logic here, i.e. only replace the link if it leads to the corresponding core activity link, and not replace it if it links to something else.